### PR TITLE
CMAKE multi threader configuration option

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by zivy — APPROVED** [review]
LGTM

File: Modules/Core/Common/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,10 @@ mark_as_advanced(CMAKE_THREAD_LIBS)
 set(ITK_DEFAULT_MAX_THREADS 128 CACHE STRING "The maximum number of threads allowed. Default: 128")
```
Comment (dzenanz) [inline Modules/Core/Common/CMakeLists.txt:24]: consitenly misspelled in commit message

File: Modules/Core/Common/src/itkConfigure.h.in [hunk 2]
```diff
@@ -170,12 +170,21 @@
...
 
-#if defined(ITK_USE_TBB)
-#define ITK_DEFAULT_THREADER TBB
-#elif defined(ITK_USE_POOL_MULTI_THREADER)
-#define ITK_DEFAULT_THREADER Pool
-#else
-#define ITK_DEFAULT_THREADER Platform
+
+#define ITK_DEFAULT_THREADER @ITK_DEFAULT_THREADER@
+#define ITK_DEFAULT_THREADER_@ITK_DEFAULT_THREADER@
```
Comment (dzenanz) [inline Modules/Core/Common/src/itkConfigure.h.in:175]: This line is almost the same as line 174. The change from space to underscore is hard to spot :)

**Review by dzenanz — APPROVED** [review]
Looks good, assuming it works 😄

File: Modules/Core/Common/src/itkConfigure.h.in (hunk 2 above)
Comment (blowekamp) [reply Modules/Core/Common/src/itkConfigure.h.in:175]: Adding comment.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/CMakeLists.txt (hunk 1 above)
Comment (dzenanz) [reply Modules/Core/Common/CMakeLists.txt:24]: consitently -> consistently :)

**Pull request merged**